### ISSUE TITLE
can't download static version

### ISSUE THREAD

**Issue by oupala** [wallabag/wallabag#3071]
### Issue details

I'm trying to download a static version of Wallabag from http://static.wallabag.org/releases/ using ansible (for automated install).

But ansible return an error:

```bash
fatal: [target]: FAILED! => {"changed": false, "failed": true, "msg": "Failure downloading http://static.wallabag.org/releases/wallabag-release-2.2.2.tar.gz, HTTP Error 403: Forbidden"}
```

I think there is a restriction on the server (maybe based on the user agent) to prevent abuse download.

### Environment

* wallabag version (or git revision) that exhibits the issue: n/a
* How did you install wallabag? Via `git clone` or by downloading the package? downloading the package
* Last wallabag version that did not exhibit the issue (if applicable): n/a
* php version: n/a
* OS: n/a
* type of hosting (shared or dedicated): n/a
* which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL): n/a

### Steps to reproduce/test case

Try to download wallabag static version using the following task:

```ansible
- name: unarchive
  unarchive:
    src: "http://static.wallabag.org/releases/wallabag-release-2.2.2.tar.gz"
    dest: "/path/to/somwhere"
    copy: no
```

**Comment by nicosomb** [comment]
Hello @oupala! 

What is your user agent? 
I just tried via `wget`, it's ok for me.

**Comment by oupala** [comment]
I don't know what is my user agent, as I'm using ansible. But you should have some logs on your server (look for: `HTTP Error 403: Forbidden`).

I also tried with wget and it works. But with ansible with the same url, your server answer 406 forbidden...

**Comment by tcitworld** [comment]
`403` or `406` ?

**Comment by oupala** [comment]
Sorry, error 403 (I edited my previous post).

**Comment by oupala** [comment]
Maybe the user agent is `ansible-httpget` (see [source file](https://github.com/dagwieers/ansible/blob/976674bb4b1eb00d2025cbdfc7b12a21c036b4cd/library/get_url#L100)).

**Comment by oupala** [comment]
You're not filtering `ansible-httpget` user agent:

```bash
wget --user-agent="ansible-httpget" http://static.wallabag.org/releases/wallabag-release-2.2.2.tar.gz--2017-05-02 21:37:15--  http://static.wallabag.org/releases/wallabag-release-2.2.2.tar.gz
Résolution de static.wallabag.org (static.wallabag.org)… 104.28.23.119, 104.28.22.119, 2400:cb00:2048:1::681c:1677, ...
Connexion à static.wallabag.org (static.wallabag.org)|104.28.23.119|:80… connecté.
requête HTTP transmise, en attente de la réponse… 200 OK
Taille : 40649644 (39M) [application/x-tar]
Sauvegarde en : « wallabag-release-2.2.2.tar.gz »
[...]
```

But you're filtering `Python-urllib/3.4` default user agent:

```bash
>wget --user-agent="Python-urllib/3.4" http://static.wallabag.org/releases/wallabag-release-2.2.2.tar.gz
--2017-05-02 21:37:56--  http://static.wallabag.org/releases/wallabag-release-2.2.2.tar.gz
Résolution de static.wallabag.org (static.wallabag.org)… 104.28.22.119, 104.28.23.119, 2400:cb00:2048:1::681c:1777, ...
Connexion à static.wallabag.org (static.wallabag.org)|104.28.22.119|:80… connecté.
requête HTTP transmise, en attente de la réponse… 403 Forbidden
2017-05-02 21:37:56 erreur 403 : Forbidden.
```

Why are you filtering this user agent? It prevent static wallabaf releases to be downloaded by ansible.

**Comment by j0k3r** [comment]
As far as I know that user-agent is often considered as bad because most of the bad crawler are build using python.

**Comment by oupala** [comment]
I've read that too. But most of the bad crawlers are run on a computer. You won't filter computers from accessible Wallabag's download site, will you?

My opinion is that using the urllib user agent as a filtering criteria is a bad criteria? You also block all legitimate usage of urllib. Python is widely used nowadays, for good and bad usages.

**Comment by j0k3r** [comment]
I've never said that there were a filter on that user agent.
I don't manage wallabag.org, @nicosomb does

**Comment by tcitworld** [comment]
If there's a filter, I guess it comes from https://github.com/DirectoryLister/DirectoryLister, not us, at least not voluntarily.

**Comment by nicosomb** [comment]
If there is filter, it comes from my web hoster, because @oupala tries to download the ZIP file directly.

**Comment by oupala** [comment]
@nicosomb Do you plan to check if there is a filter? Do you plan to ask your hoster to remove the filter if there is one?

The same problem happens if I try to download Wallabag from your [permalink](https://wllbg.org/latest-v2-package) as the permalink finally redirects to http://static.wallabag.org/releases/wallabag-release-${version}.tar.gz

**Comment by oupala** [comment]
I just checked the real user agent of ansible.

My version of ansible is using the following user agent:

```http
Python-urllib/2.7
```

And it gets filtered by the Wallabag static server.

**Comment by nicosomb** [comment]
The problem comes from Cloudflare. I will check that.

**Comment by nicosomb** [comment]
It seems that you can use https://github.com/Anorov/cloudflare-scrape

This bug is not on our side. The best way is to change your tool to download releases.

**Comment by oupala** [comment]
The bug is not on your side, nor on my side. The bug is in the garden of CloudFlare that you choosed to use.

I'm afraid I can't change the the tool I'm using for downloading Wallabag. This tool is [ansible](https://www.ansible.com/) and I use ansible to bootstrap and to deploy a whole personal server with many services.

I'm adding services one after another. I was about to add Wallabag, but I will have to drop it if it is not even possible to download it.

The bug is maybe not on your side, but it is less on my side than on your side as it is your server that is returning an http 403 error. You're (or CloudFlare is) blocking legitimate usage.

What if bot start to use Chrome or Firefox user agent? Will you let CloudFlare block Chrome of Firefox users?

**Comment by tcitworld** [comment]
FYI, since you use Ansible, you should install wallabag the recommanded way and not through the release package : http://doc.wallabag.org/en/master/user/installation.html#on-a-dedicated-web-server-recommended-way

**Comment by oupala** [comment]
@tcitworld I tried it, but Wallabag's recommended install is - in my opinion - overengineered.

It requires git, make, composer whereas the standard install would by default only require a basci ftp connection.

And I don't install softwares via git with ansible as it become almost impossible to modify any file, or the changes will be reverted the next time the playbook is run.

**Comment by nicosomb** [comment]
FYI, I plan to remove Cloudflare (only used for https), to replace it with Let's encrypt. But it's not in my urgent tasks.

**Comment by tcitworld** [comment]
> It requires git, make, composer whereas the standard install would by default only require a basci ftp connection.

Oh come on. You use ansible. This can be done in 10 minutes. It's 2017. People don't use ftp.

**Comment by tcitworld** [comment]
Use https://framabag.org/wallabag-release-2.2.2.tar.gz in the meantime.

**Comment by oupala** [comment]
@nicosomb Good news to replace CloudFlare by Let's encrypt. I will suspend my work on this subject at this tim. Could you please update this issue once you've done it?

**Comment by oupala** [comment]
@tcitworld I'm not using ftp, it was just an example to show how much the install process of Wallabag is overengineered in my opinion. There is no reason to use `make` for installing Wallabag, as `make` is only used to wrap other tools.

**Comment by oupala** [comment]
@tcitworld Ok, I'll try with https://framabag.org/wallabag-release-2.2.2.tar.gz

I'll report here if it works or not.

Are all releases available on framabag.org? And future releases? Or is it just a temporary workaround to unblock me?

**Comment by tcitworld** [comment]
> Are all releases available on framabag.org?

All the 2.x releases.

> And future releases?

Probably not.

**Comment by nicosomb** [comment]
It should be ok now: I replaced cloudflare with let's encrypt. Can you tell me if it's OK for you please.

**Comment by oupala** [comment]
It is ok, it works well now. Thanks for quitting CloudFlare !

Were you using CloudFlare to enable TLS on your website?

**Comment by oupala** [comment]
The first problem is fixed and I have now found another problem: the directory name after unarchiving the archive.

The name of the directory after unarchiving the archive is `release-2.2.2` which is not very helpful to know what it is about. I think it should name something like `wallabag-2.2.2`.

Tell me if you prefer me to open another issue instead of using the same following issue.
